### PR TITLE
Include namespaced brokers in svcat get brokers

### LINKED ISSUE #2218: Include namespace-scoped brokers in svcat get brokers
You may want to look at https://github.com/kubernetes-incubator/service-catalog/pull/2116 where I'm making the same changes for classes for an example.

```
$ svcat get brokers [--namespace] [--all-namespaces]
  NAME     NAMESPACE   URL.      STATUS
+---------+----------+---------+--------+
broker1               http://... Ready
broker2    ns-test    http://... Ready
```

* Cluster scoped brokers are always listed
* The namespaced interface should be implemented

### PR DESCRIPTION
**Pull request by carolynvs**
```
$ svcat get brokers
     NAME      NAMESPACE                              URL                              STATUS
+------------+-----------+-----------------------------------------------------------+--------+
  minibroker               http://minibroker-minibroker.minibroker.svc.cluster.local   Ready

$ svcat get brokers -n minibroker
      NAME       NAMESPACE                               URL                              STATUS
+--------------+------------+-----------------------------------------------------------+--------+
  minibroker                  http://minibroker-minibroker.minibroker.svc.cluster.local   Ready
  myminibroker   minibroker   http://minibroker-minibroker.minibroker.svc.cluster.local   Ready

$ svcat get brokers --scope=cluster
     NAME      NAMESPACE                              URL                              STATUS
+------------+-----------+-----------------------------------------------------------+--------+
  minibroker               http://minibroker-minibroker.minibroker.svc.cluster.local   Ready

$ svcat get brokers -n minibroker --scope=namespace
      NAME       NAMESPACE                               URL                              STATUS
+--------------+------------+-----------------------------------------------------------+--------+
  myminibroker   minibroker   http://minibroker-minibroker.minibroker.svc.cluster.local   Ready

$ svcat get brokers --all-namespaces
         NAME          NAMESPACE                               URL                              STATUS
+--------------------+------------+-----------------------------------------------------------+--------+
  minibroker                        http://minibroker-minibroker.minibroker.svc.cluster.local   Ready
  another-minibroker   default      http://minibroker-minibroker.minibroker.svc.cluster.local
  myminibroker         minibroker   http://minibroker-minibroker.minibroker.svc.cluster.local   Ready
```
Preview of the docs available at: https://deploy-preview-2227--svc-cat.netlify.com/docs/cli/#find-brokers-installed-on-the-cluster

TODO:
- [x] Add unit tests for RetrieveBrokers call
- [x] Add unit tests for brokers svcat command
- [x] Update docs with new behavior and flags

Closes #2218.

### CODE REVIEW

File: cmd/svcat/command/scoped.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import (
+	"fmt"
+
+	"errors"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/svcat/service-catalog"
+	"github.com/spf13/pflag"
+)
+
+// HasScopedFlags represents a command that can be scoped to can be scoped
```
Comment (jboyd01) [inline cmd/svcat/command/scoped.go]: duplicate words "can be scoped to"

**Comment by MHBauer** [comment]
@carolynvs I dug the details out of gcs that was pointed out to me
```

I0726 17:16:25.927] === RUN   TestUpdateServiceInstanceNewDashboardResponse
I0726 17:16:25.928] === RUN   TestUpdateServiceInstanceNewDashboardResponse/alpha_features_enabled
W0726 17:16:37.472] 2018-07-26 17:16:37.471528 W | wal: sync duration of 1.359038397s, expected less than 1s
W0726 17:16:38.140] 2018-07-26 17:16:38.139553 W | etcdserver: apply entries took too long [368.242626ms for 1 entries]
W0726 17:16:38.140] 2018-07-26 17:16:38.139640 W | etcdserver: avoid queries with large range/delete range!
W0726 17:17:31.257] E0726 17:17:31.256098       1 controller_instance.go:1685] ServiceInstance "test-namespace/test-instance" v842: Failed to update references: Timeout: request did not complete within allowed duration
I0726 17:17:34.354] === RUN   TestUpdateServiceInstanceNewDashboardResponse/alpha_feature_disabled
W0726 17:17:37.052] E0726 17:17:37.051306       1 controller_instance.go:1685] ServiceInstance "test-namespace/test-instance" v842: Failed to update references: Operation cannot be fulfilled on serviceinstances.servicecatalog.k8s.io "test-instance": the object has been modified; please apply your changes to the latest version and try again
W0726 17:17:41.405] 2018-07-26 17:17:41.405036 W | wal: sync duration of 1.599916754s, expected less than 1s
W0726 17:17:53.538] 2018-07-26 17:17:53.537639 W | etcdserver: apply entries took too long [276.964261ms for 1 entries]
W0726 17:17:53.538] 2018-07-26 17:17:53.537740 W | etcdserver: avoid queries with large range/delete range!
W0726 17:17:56.325] 2018-07-26 17:17:56.324981 W | wal: sync duration of 1.114856277s, expected less than 1s
W0726 17:18:00.381] 2018-07-26 17:18:00.381358 W | wal: sync duration of 1.122258947s, expected less than 1s
W0726 17:18:03.780] 2018-07-26 17:18:03.779714 W | wal: sync duration of 1.416115309s, expected less than 1s
I0726 17:18:08.411] --- FAIL: TestUpdateServiceInstanceNewDashboardResponse (102.58s)
I0726 17:18:08.411]     --- FAIL: TestUpdateServiceInstanceNewDashboardResponse/alpha_features_enabled (68.53s)
I0726 17:18:08.411]     	framework.go:111: Server started on port 36265
I0726 17:18:08.411]     	framework.go:142: Test client will use API Server URL of https://localhost:36265
I0726 17:18:08.412]     	controller_test.go:790: controller start
I0726 17:18:08.412]     	controller_test.go:1200: error waiting for instance to become ready: timed out waiting for the condition
I0726 17:18:08.412]     --- PASS: TestUpdateServiceInstanceNewDashboardResponse/alpha_feature_disabled (34.06s)
I0726 17:18:08.412]     	framework.go:111: Server started on port 43229
I0726 17:18:08.412]     	framework.go:142: Test client will use API Server URL of https://localhost:43229
I0726 17:18:08.412]     	controller_test.go:790: controller start
I0726 17:18:08.412]     	framework.go:116: Shutting down server on port: 43229
```

**Comment by carolynvs** [comment]
@MHBauer I didn't make any changes that would have affected those tests. The only non-svcat changes were adding methods like `GetName()` to SerivceBroker and ClusterServiceBroker.

I'm not quite sure if this is something that is a problem with just my PR that I need to try to figure out how to see what gcp is doing, or if other PRs are also rando hitting the the same problem? I may just wait and see if others hit this before deciding that I caused this, and need to change the PR to fix it. 😀

**Comment by MHBauer** [comment]
could be a flake then. oh noes.

**Comment by carolynvs** [comment]
@MHBauer I'll open an issue for that, seems to be a flake as I just pushed up a doc commit and now the test is passing. 😝

**Pull request merged**